iPetCare/Application/Services/ExaminationTypesService.cs

```diff
@@ -0,0 +1,167 @@
+﻿using System;
+using System.Linq;
+using System.Collections.Generic;
+using System.Net;
+using Application.Interfaces;
+using Application.Services.Utilities;
+using Application.Dtos.ExaminationTypes;
+using Domain.Models;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.EntityFrameworkCore;
+using Persistence;
+
+namespace Application.Services
+{
+    public class ExaminationTypesService : Service, IExaminationTypes
+    {
+        public ExaminationTypesService(IServiceProvider serviceProvider) : base(serviceProvider)
+        {
+        }
+
+        public async Task<ServiceResponse<ExaminationTypesCreateDtoResponse>> CreateExaminationTypesAsync(ExaminationTypesCreateDtoRequest dto)
+        {
+            if (await Context.ExaminationTypes.Where(x => x.Name == dto.Name).AnyAsync())
+                return new ServiceResponse<ExaminationTypesCreateDtoResponse>(HttpStatusCode.BadRequest, "Podane badanie już istanieje");
+
+            var currentUserName = CurrentlyLoggedUserName;
+            if (currentUserName != null)
```

Comment:
Ogólnie aby uniknąć zbytniego zagnieżdżania proponuję sprawdzać czy currentusername == null i jeśli tak, to od razu wyjść z metody z odpowiednim responsem - czysto estetyczne ofc.